src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+extern alias WORKSPACES;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.CodeAnalysis.Editor.Interactive
+{
+    internal sealed class NuGetPackageResolverImpl : WORKSPACES::Microsoft.CodeAnalysis.Scripting.NuGetPackageResolver
+    {
+        private readonly string _packagesDirectory;
+
+        internal NuGetPackageResolverImpl(string packagesDirectory)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(packagesDirectory));
+            Debug.Assert(PathUtilities.IsAbsolute(packagesDirectory));
+
+            _packagesDirectory = packagesDirectory;
+        }
+
+        internal override ImmutableArray<string> ResolveNuGetPackage(string reference)
+        {
+            string packageName;
+            string packageVersion;
+            if (!ParsePackageReference(reference, out packageName, out packageVersion))
+            {
+                return default(ImmutableArray<string>);
+            }
+
+            var tempPath = PathUtilities.CombineAbsoluteAndRelativePaths(Path.GetTempPath(), Guid.NewGuid().ToString("D"));
+            var tempDir = Directory.CreateDirectory(tempPath);
+            try
+            {
+                // Create project.json.
+                var projectJson = PathUtilities.CombineAbsoluteAndRelativePaths(tempPath, "project.json");
+                using (var stream = File.OpenWrite(projectJson))
+                {
+                    using (var writer = new StreamWriter(stream))
```

Comment:
Convention is to omit the first `{` and indent the multiple `using`s the same level.